src/main/java/org/jenkinsci/plugins/codesonar/services/CodeSonarCacheService.java

```diff
@@ -14,12 +14,14 @@
 import org.jenkinsci.plugins.codesonar.models.CodeSonarAnalysisData;
 import org.jenkinsci.plugins.codesonar.models.CodeSonarAnalysisWarningCount;
 import org.jenkinsci.plugins.codesonar.models.CodeSonarHubInfo;
+import org.jenkinsci.plugins.codesonar.models.CodeSonarWarningCount;
 import org.jenkinsci.plugins.codesonar.models.analysis.Analysis;
 import org.jenkinsci.plugins.codesonar.models.metrics.Metrics;
 import org.jenkinsci.plugins.codesonar.models.procedures.ProcedureMetric;
 import org.jenkinsci.plugins.codesonar.models.procedures.Procedures;
 
 public class CodeSonarCacheService {
+    private static CodeSonarCacheService instance = null;

```

Comment:
There a single instance of this class that has to be kept during a whole pipeline execution lifecycle, and it's common to keep that instance on the class itself as a static variable.

Reply:
The truth is that the plugin can run simultaneously on different pipelines, we should probably consider stop using the singleton pattern and let each `CodeSonarPublisher `class hold the specific instance of the `CodeSonarCacheService`, making it responsible for passing that isntance to those Jenkins conditions that need it in order to retrieve further data.